### PR TITLE
Add the instructions for the trusted setup

### PR DESCRIPTION
**Pull request by xevisalle**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,15 @@
-# trusted-setup
\ No newline at end of file
+# Trusted Setup for BLS12-381
```
Comment (marta-belles) [inline README.md]: Trusted Setup -> Trusted **s**etup (to make it consisent with the rest of section titles).

File: README.md [hunk 2]
```diff
@@ -1 +1,15 @@
...
+
+This repository contains a multi-party computation (MPC) ceremony to compute the powers of tau needed by different proving systems, using the BLS12-381 setting. In particular, we compute up to 2**21 powers of tau, taking as a starting point the ceremony performed by Zcash, that can be found [here](https://github.com/ZcashFoundation/powersoftau-attestations/tree/master).
```
Comment (marta-belles) [inline README.md]: This repository contains a multi-party computation (MPC) ceremony to compute the powers of tau needed by different ~~proving systems using the BLS12-381 setting~~ **proof systems using the BLS12-381 elliptic curve construction**. In particular, we compute up to 2**21 powers of tau, taking as a starting point the ceremony performed by Zcash, that can be found here. 

Justification:
- I know Sean writes _proving_ but Javi always said _proof_ systems.
- It is better to add the word _elliptic curve_, to clarify what we are doing.

File: README.md [hunk 3]
```diff
@@ -1 +1,15 @@
...
+
+## The ceremony
+
+First, an open call for participation will be made through Dusk socials. For transparency purposes, some information will be required (real name, link to social network profile, etc.). Then, a Dusk team member will be chosen as a coordinator for the ceremony. They will verify the contributions of Zcash up to number 87, and will create a new challenge following these [instructions](instructions/VERIFY.md).
```
Comment (marta-belles) [inline README.md]: Then, a Dusk team member will be chosen as a coordinator for the ceremony. ~~They~~ **The coordinator** will verify the contributions of Zcash up to number 87, and will create a new challenge following these instructions.

Justification:
- Although _They_ is correct, it is less clear who is doing the verifications.

File: README.md [hunk 4]
```diff
@@ -1 +1,15 @@
...
+
+The list of participants will be sorted, and the coordinator will contact the first participant providing the previously computed challenge. The first participant will compute their contribution using these [instructions](instructions/CONTRIBUTE.md), and will send the response back to the coordinator. The coordinator will follow these [instructions](instructions/COORDINATOR.md) to verify the contribution and its integrity. In this step, a new challenge is computed, and the same process is repeated with the next participant.
```
Comment (marta-belles) [inline README.md]: In this step, ~~a new challenge is computed~~ **the coordinator computes a new challenge**, and the same process is repeated with the next participant. 

Justification:
- Make it more clear who does what.

File: README.md [hunk 5]
```diff
@@ -1 +1,15 @@
...
+
+All the contributions will be available in the folder `contributions`.
```
Comment (marta-belles) [inline README.md]: - All ~~the~~ contributions will be available in the folder `contributions`.
- If possible, I would create already a folder named `contributions` and link it here.

File: instructions/CONTRIBUTE.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+## Compute your contribution
+
+First, clone the following code:
+
+```
+git clone https://github.com/dusk-network/powersoftau.git
+```
+
+Once you get contacted by the coordinator, you will be provided with a challenge file `challenge`. Place the challenge in the root of the cloned repository, enter the repository, and execute:
```
Comment (marta-belles) [inline instructions/CONTRIBUTE.md]: [...] with a challenge file **named** `challenge`. Place ~~the challenge~~ **the file** in the root of the cloned **repository. Navigate to the repository and execute:**

File: instructions/CONTRIBUTE.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+```
+cargo run --release --bin compute
+```
+
+When executing the previous command, you will be asked for entropy. It is important that you introduce a quite large and completely random input. Some examples of entropy sources are:
```
Comment (marta-belles) [inline instructions/CONTRIBUTE.md]: I suggest saying:

It is important that you introduce a ~~quite~~ large and completely random input **for optimal security**.

File: instructions/CONTRIBUTE.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+- Randomly typing on your keyboard with the eyes closed.
+- Take random words from random pages in random books or magazines you have at home.
+- Generate various values using `/dev/urandom` in different devices.
+
+Such input **does not have to** be copied or stored anywhere.
```
Comment (marta-belles) [inline instructions/CONTRIBUTE.md]: Such input ~~does not have to~~ **must not** be copied or stored anywhere.

Justification:
- _Does not have to_ means it is not necessary, but here we want to stress that it is something that _should not be done_.

File: instructions/CONTRIBUTE.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+It will take some time to compute a `response` file containing your contribution. Once finished, you will need to submit your contribution.
+
+## Submit your contribution
+
+To submit your contribution, provide to the coordinator what follows:
```
Comment (marta-belles) [inline instructions/CONTRIBUTE.md]: To submit your contribution, provide **the following to the coordinator**:

File: instructions/CONTRIBUTE.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+
+- The generated `response` file.
+- A `report` file containing the following information:
```
Comment (marta-belles) [inline instructions/CONTRIBUTE.md:37]: Personally, I think that adding the **time taken** may be a good idea (@HDauven, do you think that is something contributors will like to add?).

File: instructions/CONTRIBUTE.md [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+  - Brief description of the device used.
+  - The hash of the challenge you used.
+  - The hash of the response file.
+  - Brief description of the entropy source you used and how you ensured that it's not possible to recover the value you introduced.
```
Comment (marta-belles) [inline instructions/CONTRIBUTE.md]: - Brief description of the entropy source you used and **any precautions you took to ensure** ~~how you ensured~~ that it's not possible to recover the **random** value you introduced.

Justification:
- _how you ensured_ sounds very strong. I think that we should allow people to make their contributions even if they didn't take extreme precautions. I believe the new phrasing helps in that sense.

File: instructions/COORDINATOR.md [hunk 12]
```diff
@@ -0,0 +1,21 @@
+
+# Coordinate the ceremony
+
+## Prerequisites
+
+You will need Rust to compile the code that verifies the contributions. You can follow the recommended [guide](https://www.rust-lang.org/tools/install).
+
+## Coordinate the participants
+
+First, verify the [contributions](https://github.com/dusk-network/powersoftau-attestations) of Zcash up to number 87 using this [guide](VERIFY.md). This will output a challenge file you need to send to the first participant on the list.
```
Comment (marta-belles) [inline instructions/COORDINATOR.md]: ~~This~~ **The last step of this verification** will output a challenge file you need to send to the first participant on the list.

File: instructions/COORDINATOR.md [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+
+Once you get the submission from the participant, you will need to do the following:
+
+- Check the identity of the participant by verifying the provided signature or the provided post on a social network.
+- Check the integrity of the `response` file using the provided hash.
+- Verify that the contribution has been performed using the correct challenge. To do so, place the response from the participant into the root directory of the the code, where you must have the challenge you sent them, and verify as follows:
```
Comment (marta-belles) [inline instructions/COORDINATOR.md]: - Verify that the contribution has been performed using the correct challenge. To do so, **first** place the response from the participant into the root directory of the the code, where you must **also** have the challenge you sent them. **Then, verify** as follows:

File: instructions/COORDINATOR.md [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+
+```
+cargo run --release --bin verify_transform 
+```
+Once verified, a new file `new_challenge` will be generated. Rename it to `challenge` and send it to the next participant on the list.
```
Comment (marta-belles) [inline instructions/COORDINATOR.md]: Once **the contribution has been** verified, **the same instruction will generate** a new file **named** `new_challenge` ~~will be generated~~. Rename it to `challenge` and send it to the next participant on the list.

Remark: I assume the same instruction already produces the new challenge, right? Or should the contribution run a different command to produce the new challenge?

File: instructions/VERIFY.md [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+git clone https://github.com/dusk-network/powersoftau.git
+```
+
+Download the first contribution `response`, place it into the root of the cloned repository, and execute:
+
+```
+cargo run --release --bin new # this generates the first challenge
+cargo run --release --bin verify_transform # this verifies that the response has been computed using the given challenge
+```
+Once verified, a new file `new_challenge` will be generated. Now you can download the next response file, and keep verifying all the contributions as follows:
```
Comment (marta-belles) [inline instructions/VERIFY.md]: Same comment as before:

Once **the contribution has been** verified, **the same instruction will generate** a new file **named** `new_challenge` ~~will be generated~~. 

Remark: I assume the same instruction already produces the new challenge, right? Or should the contribution run a different command to produce the new challenge?

**Review by marta-belles — CHANGES_REQUESTED** [review]
I like the flow, but I would make some minor changes to the text to clarify the steps. Here are my suggestions.

File: README.md (hunk 3 above)
Comment (HDauven) [reply README.md]: Are we sure it is 87 contributions by the way @marta-belles @xevisalle?

I've also verified the response of [Random Beacon: 88](https://github.com/dusk-network/powersoftau-attestations/tree/master/0088).

File: instructions/CONTRIBUTE.md (hunk 10 above)
Comment (HDauven) [reply instructions/CONTRIBUTE.md:37]: > Personally, I think that adding the **time taken** may be a good idea (@HDauven, do you think that is something contributors will like to add?).

If you think it is a valuable data point, sure.

File: README.md (hunk 3 above)
Comment (xevisalle) [reply README.md]: Yes, the random beacon is not necessary for this reason: https://github.com/filecoin-project/phase2-attestations#phase1-randomness-beacon . However, we can always apply the random beacon at the end.

File: README.md (hunk 5 above)
Comment (xevisalle) [reply README.md]: I think it won't show up until it's filled.

File: instructions/COORDINATOR.md (hunk 14 above)
Comment (xevisalle) [reply instructions/COORDINATOR.md]: yes

File: instructions/VERIFY.md (hunk 15 above)
Comment (xevisalle) [reply instructions/VERIFY.md]: yes

**Review by HDauven — APPROVED** [review]

**Review by marta-belles — APPROVED** [review]

**Pull request merged**